doc/plutus/architecture/constraints/index.rst

```diff
@@ -0,0 +1,294 @@
+          }
+
+  data ScriptOutputConstraint a =
+      ScriptOutputConstraint
+          { ocDatum :: a -- ^ Typed datum to be used with the target script
+          , ocValue :: Value
+          }
+
+The `TxConstraints` can be combined with its `Semigroup` instance.
+Values of `TxConstraints` are constructed using smart constructors like `mustBeSignedBy`, `mustPayToTheScript`, `mustSpendAtLeast`, etc. which are available in Ledger.Constraints.TxConstraints_.
+
+.. code-block:: haskell
+
+  checkScriptContext
+      :: forall i o. ToData o
+      => TxConstraints i o
+      -> ScriptContext
+      -> Bool
+
+  data ScriptLookups a =
+      ScriptLookups
+          { slMPS                  :: Map MintingPolicyHash MintingPolicy
+          -- ^ Minting policies that the script interacts with
+          , slTxOutputs            :: Map TxOutRef ChainIndexTxOut
+          -- ^ Unspent outputs that the script may want to spend
+          , slOtherScripts         :: Map ValidatorHash Validator
+          -- ^ Validators of scripts other than "our script"
+          , slOtherData            :: Map DatumHash Datum
+          -- ^ Datums that we might need
+          , slPaymentPubKeyHashes  :: Map PaymentPubKeyHash PaymentPubKey
+          -- ^ Public keys that we might need
+          , slTypedValidator       :: Maybe (TypedValidator a)
+          -- ^ The script instance with the typed validator hash & actual compiled program
+          , slOwnPaymentPubKeyHash :: Maybe PaymentPubKeyHash
+          -- ^ The contract's payment public key hash, used for depositing tokens etc.
+          , slOwnStakePubKeyHash   :: Maybe StakePubKeyHash
+          -- ^ The contract's stake public key hash (optional)
+          } deriving stock (Show, Generic)
+            deriving anyclass (ToJSON, FromJSON)
+
+  mkTx
+      :: ( FromData (DatumType a)
+         , ToData (DatumType a)
+         , ToData (RedeemerType a)
+         )
+      => ScriptLookups a
+      -> TxConstraints (RedeemerType a) (DatumType a)
+      -> Either MkTxError UnbalancedTx
+
+Example
+-------
+
+Here's a complete example using constraints.
+We want to create a Plutus app that locks some Ada in a script output and splits them evenly between two recipients.
+
+Let's start by declaring the required imports.
+
+.. literalinclude:: BasicAppConstraints.hs
+   :start-after: BLOCK0
+   :end-before: BLOCK1
+
+We will then declare the ``SplitData`` datatype which describes the two recipients of the funds, and the total amount of the funds denoted in Ada.
+
+.. literalinclude:: BasicAppConstraints.hs
+   :start-after: BLOCK1
+   :end-before: BLOCK2
+
+Based on this datatype, we will start by creating a function which will generate constraints for:
+
+1- the validator script which will validate the spending of it's script output based on ``SplitData``
+
+2- the transaction we will create to spend these script outputs
+
+.. literalinclude:: BasicAppConstraints.hs
```

Comment:
is this implementing 1 or 2 above? I like the itemised approach but it might need more headings to be clear.